### PR TITLE
[BIOMAGE-1539] Fix handling of large work results

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1539

#### Link to staging deployment URL 
https://ui-martinfosco-ui538-worker20.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/worker/pull/206

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Code updates
Have best practices and ongoing refactors being observed in this PR

- [ ] Migrated any selector / reducer used to the new format
- [ ] Validated that current unit tests work as expected and are enough

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging
- [ ] Passed integration tests 

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Comment by cosa65** [comment]
The actual problem turned out to be not that the `markerGenes` json response to parse was 500mB+ but that the R worker's response is stringified and wrapped in a new json.

In the UI is was necessary to do first a `response.json()` and later on a `JSON.parse(response.results[0].body)` to get the R worker's data.

The problem was that, during the first json parse: `response.json()`, nodejs found the actual task's response (that is stored as a string) and in very big responses this string is bigger than the string length limit set in nodejs, so it would error out.

To fix this at first I considered parsing the R worker response in the python worker before storing it in s3, but it seems that the whole structure we are using to store the work response is not used at all since in the UI we always do `response.results[0].body`, si I decided to move this unpacking to the python worker (where the error/cacheable properties are last checked) and just directly store in s3 the json response.

So now the UI just gets what it needs so that there isn't a stringified json inside another json (and we also avoid having to do 2 json parses for each response).

File: src/__test__/utils/work/seekWorkResponse.test.js [hunk 1]
```diff
@@ -98,8 +92,8 @@ describe('seekFromAPI unit tests', () => {
 
     expect(socketConnectionMocks.mockOn).toHaveBeenCalledTimes(1);
     expect(response).toEqual({
-      results: [{ body: '{"hello":"world"}' }],
-      response: { error: false },
```
Comment (aerlaut) [inline src/__test__/utils/work/seekWorkResponse.test.js:102]: Is it ok to remove the error flag?

**Review by ivababukova — APPROVED** [review]

File: src/__test__/utils/work/seekWorkResponse.test.js (hunk 1 above)
Comment (cosa65) [reply src/__test__/utils/work/seekWorkResponse.test.js:102]: As far as I could see, in the UI we don't use the error flag at all

Comment (cosa65) [reply src/__test__/utils/work/seekWorkResponse.test.js:102]: In fact, if you look at `def publish(self):` in the python worker, it calls `__upload(self):` (which performs the upload to s3), but it calls `_upload` only if error is false

Comment (aerlaut) [reply src/__test__/utils/work/seekWorkResponse.test.js:102]: ok, so error is not used in the UI 👍

**Pull request merged**